pkg/controller/pd_control.go

```diff
@@ -21,8 +21,14 @@ import (
 // GetPDClient gets the pd client from the TidbCluster
 func GetPDClient(pdControl pdapi.PDControlInterface, tc *v1alpha1.TidbCluster) pdapi.PDClient {
 	if tc.IsHeterogeneous() {
+		if len(tc.Spec.ClusterDomain) > 0 {
+			return pdControl.GetClusterRefPDClient(pdapi.Namespace(tc.GetNamespace()), tc.Spec.Cluster.Name, tc.Spec.ClusterDomain, tc.IsTLSClusterEnabled())
```

Comment:
tc.Spec.ClusterDomain should be `tc.Spec.Cluster.ClusterDomain` ?

Reply:
It seems @july2993 is correct, @handlerww could you please double confirm?

Reply:
`tc.Spec.ClusterDomain` is the local ClusterDomain. `tc.Spec.Cluster.ClusterDomain` is the ClusterDomain of the cluster you want to join.

Reply:
@july2993 is correct, tc.Spec.Cluster.Name should corresponse to tc.Spec.Cluster.ClusterDomain.

Reply:
@handlerww OK, please help submit a PR to fix this and please also check if any similar issues in the other code blocks.